src/ngx_http_lua_log_ringbuf.c

```diff
@@ -0,0 +1,203 @@
+
+#ifndef DDEBUG
+#define DDEBUG 0
+#endif
+#include "ddebug.h"
+
+
+#include "ngx_http_lua_common.h"
+#include "ngx_http_lua_log_ringbuf.h"
+
+
+typedef struct {
+    unsigned    len:24;
+    unsigned    log_level:4;
+    unsigned    is_data:1;
```

Comment:
Should we use a similar concept of `sentinel` to `ngx_queue_t` and `ngx_rbtree_t` in the nginx core? Use of the `is_data` flag in every node might not be so efficient. I posted this comment in my previous rounds of review but it seems it got lost.